src/material/card/testing/card-harness.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessLoader, HarnessPredicate} from '@angular/cdk/testing';
+import {CardHarnessFilters} from './card-harness-filters';
+
+
+/** Harness for interacting with a standard mat-card in tests. */
+export class MatCardHarness extends ComponentHarness {
+  /** The selector for the host element of a `MatCard` instance. */
+  static hostSelector = 'mat-card';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatCardHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which card instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: CardHarnessFilters = {}): HarnessPredicate<MatCardHarness> {
+    return new HarnessPredicate(MatCardHarness, options)
+        .addOption('text', options.text,
+            (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text))
+        .addOption('title', options.title,
+            (harness, title) => HarnessPredicate.stringMatches(harness.getTitleText(), title))
+        .addOption('subtitle', options.subtitle,
+            (harness, subtitle) =>
+                HarnessPredicate.stringMatches(harness.getSubtitleText(), subtitle));
+  }
+
+  private _title = this.locatorForOptional('.mat-card-title');
+  private _subtitle = this.locatorForOptional('.mat-card-subtitle');
+
+  /** Gets all of the card's content as text. */
+  async getText(): Promise<string> {
+    return (await this.host()).text();
+  }
+
+  /** Gets the cards's title text. */
+  async getTitleText(): Promise<string> {
+    return (await this._title())?.text() ?? '';
+  }
+
+  /** Gets the cards's subtitle text. */
+  async getSubtitleText(): Promise<string> {
+    return (await this._subtitle())?.text() ?? '';
+  }
+
+  /** Gets a harness loader for the header section of this card. */
+  async getHarnessLoaderForHeader(): Promise<HarnessLoader | null> {
```

Comment:
Maybe a radical idea, but what if `MatCardHarness ` _implemented_ `HarnessLoader`? This would let people do things like
```ts
await loader.getHarness(cardHarness.with({title: 'Shiba Inu'}))
    .getHarness(MatCheckboxHarness.with({label: 'Agree'});
```
vaguely like `querySelector`

I would _guess_ that most of the time when people want to grab an element out of the card, they don't strictly care which section it's in. If we like this idea, it could apply to other content-container components.

Reply:
Oooh I like that idea. Do you think it's still worth having the more specific methods? I feel like it might be, in case someone wants to e.g. grab all of the buttons in the actions section, ignoring whatever might be in the content section. For simpler components where content can only be projected into one area, we probably wouldn't need any additional methods

Reply:
I think it makes sense to target different areas, but I'm wondering if the API can be more concise. Maybe something that creates the locator internally

Reply:
Yeah, after implementing it I realized that these do feel kind of overkill. All we really need is a string with the right selector to pass to `CardHarness.getChildLoader`. I could create an enum (either something standalone like `MatCardSections.HEADER` or attached to the harness `MatCardHarness.Sections.HEADER`), then the final API would look something like this:

```ts
const card = await loader.getHarness(MatCardHarness);
const headerLoader = await card.getChildLoader(MatCardSection.HEADER);
const headerButton = await headerLoader.getHarness(MatButtonHarness);
```

Reply:
Would it work if we did something like this?
```ts
const card = await loader.getHarness(MatCardHarness);

card.getHarness(MatButtonHarness); // get a button anywhere in the card
card.getHarnessWithinHeader(MatButtonHarness); // get button from the header only
card.getHarnessWithinContent(MatButtonHarness); // get button from the content only
```

Reply:
Do you mean in addition to having the `getChildLoader` method or instead of? I think if we're going to go the route of having extra methods they might as well return `HarnessLoader` instances since those are more flexible (since they let you create child loaders rooted at arbitrary sub-elements)

Reply:
Either way. My main goal was to make the things people most commonly want to do easier / more concise. I suspect that just wanting to query for a harness from the card in general will be the most common by far, so I don't feel that strongly about the other methods